crates/bevy_sprite/src/texture_atlas_builder.rs

```diff
@@ -9,36 +9,56 @@ use rectangle_pack::{
 };
 use thiserror::Error;
 
+#[derive(Debug, Error)]
+pub enum TextureAtlasBuilderError {
+    #[error("could not pack textures into an atlas within the given bounds")]
+    NotEnoughSpace,
+}
+
 #[derive(Debug)]
+/// A builder which is used to create a texture atlas from many individual
+/// sprites.
 pub struct TextureAtlasBuilder {
-    pub textures: Vec<Handle<Texture>>,
-    pub rects_to_place: GroupedRectsToPlace<Handle<Texture>>,
-    pub initial_size: Vec2,
-    pub max_size: Vec2,
+    /// The grouped rects which must be placed with a key value pair of a
+    /// texture handle to an index.
+    rects_to_place: GroupedRectsToPlace<Handle<Texture>>,
+    /// The sprite size in pixels.
+    initial_size: Vec2,
+    /// The absolute maximum size of a sprite sheet in pixels.
+    max_size: Vec2,
 }
 
 impl Default for TextureAtlasBuilder {
     fn default() -> Self {
-        Self::new(Vec2::new(256., 256.), Vec2::new(2048., 2048.))
+        Self {
+            rects_to_place: GroupedRectsToPlace::new(),
+            initial_size: Vec2::new(256., 256.),
+            max_size: Vec2::new(2048., 2048.),
+        }
     }
 }
 
-#[derive(Debug, Error)]
-pub enum RectanglePackError {
-    #[error("Could not pack textures into an atlas within the given bounds")]
-    NotEnoughSpace,
-}
+pub type TextureAtlasBuilderResult<T> = Result<T, TextureAtlasBuilderError>;
 
 impl TextureAtlasBuilder {
-    pub fn new(initial_size: Vec2, max_size: Vec2) -> Self {
-        Self {
-            textures: Default::default(),
-            rects_to_place: GroupedRectsToPlace::new(),
-            initial_size,
-            max_size,
-        }
+    /// Constructs a new texture atlas builder.
+    pub fn new() -> Self {
```

Comment:
I rarely stray from "normal rust" conventions, but I'm not a huge fan of `new()` aliasing to `default()`. I prefer "one way" of doing things and new() vs default() creates an arbitrary choice that users need to make (where it doesn't matter one way or the other). I'd prefer it if we just removed new() entirely in favor of `default()`.

Reply:
I agree that `new` can an be arbitrary choice for some types, but it isn't always. For types that benefit from being called in a const context, `const fn new` can provide this for types with private fields, while `default` and other trait methods cannot be called (yet); however, that doesn't apply to `TextureAtlasBuilder`.

It can also help discoverability in docs/autocomplete. While I am very comfortable with grokking trait impls on types, it's a skill that takes time for new rust users, and a `new` can be presented near the top of the page. Thankfully, RA is pretty great at autocompleting `default` these days though.

Reply:
Yeah id definitely make exceptions for `const fn new`. Docs/autocomplete discoverability is a fair point, but learning to look for default() is a useful skill too :smile:

I still personally think that "less is better" here, but if enough people think "new + default everywhere" is the better policy, I'm open to reconsidering. For now lets remove it (because I've been encouraging "default only" elsewhere and have generally followed that pattern). But feel free to form a new() coalition / start a github discussion. If we land on "new() + default()" we can adapt the whole codebase.

Reply:
Haha, less sounds good here. I rarely define `new` for my own types anyway, and the `default` pattern is consistent throughout bevy. A month ago I was curious what adding a ` fn default` to `bevy_utils` would look like https://github.com/bevyengine/bevy/compare/master...memoryruins:default (then life happened and I forgot about the branch 😅)

Reply:
Whoaa thats cool. I'm still sorting out my feelings about it, but it certainly cuts down on boilerplate.